### PR TITLE
authorization header / app navigation

### PR DESCRIPTION
**Pull request by LorenaL9**

### CODE REVIEW

File: QuizApp/NetworkService/NetworkClient.swift [hunk 1]
```diff
@@ -59,16 +65,26 @@ class NetworkClient: NetworkClientProtocol {
...
 
+        let accessToken = getAccessToken()
+
         request.httpMethod = method.rawValue
         request.allHTTPHeaderFields = header
 
+        if let accessToken = accessToken {
+            request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
+        }
+
```
Comment (dorastipkovic) [inline QuizApp/NetworkService/NetworkClient.swift:76]:
```suggestion
        if let accessToken = getAccessToken() {
            request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
        }

```

File: QuizApp/Leaderboard/Client/LeaderboardClient.swift [hunk 2]
```diff
@@ -8,11 +8,10 @@ class LeaderboardClient: LeaderboardClientProtocol {
         self.networkClient = networkClient
     }
 
-    func fetchLeaderboard(token: String, quizId: Int) async -> Result<[LeaderboardResponse], RequestError> {
-        let header = ["Authorization": "Bearer \(token)"]
+    func fetchLeaderboard(quizId: Int) async -> Result<[LeaderboardResponse], RequestError> {
         let query = [URLQueryItem(name: "quizId", value: String(quizId))]
 
-        return await networkClient.get(header: header, body: nil, path: .quizLeaderboard, query: query)
+        return await networkClient.get(header: nil, body: nil, path: .quizLeaderboard, query: query)
```
Comment (dorastipkovic) [inline QuizApp/Leaderboard/Client/LeaderboardClient.swift:14]: Make `header` argument have default value `nil`, like `query` does.

**Review by dorastipkovic — CHANGES_REQUESTED** [review]
Don't have two separate topics covered in one PR. 
Authorization header should have been one PR, and navigation the other. 
Leave it as is now, but please follow that principle in the future.

File: QuizApp/Leaderboard/Client/LeaderboardClient.swift (hunk 2 above)
Comment (LorenaL9) [reply QuizApp/Leaderboard/Client/LeaderboardClient.swift:14]: I made the same for **body** argument, but only for functions that are using in some client.

**Review by dorastipkovic — APPROVED** [review]

**Pull request merged**